contrib/soju/template.py

```diff
@@ -0,0 +1,74 @@
+pkgname = "soju"
+pkgver = "0.7.0"
+pkgrel = 0
+build_style = "makefile"
+hostmakedepends = ["go", "scdoc"]
+makedepends = ["sqlite-devel"]
+depends = ["ca-certificates", "libcap-progs"]
+pkgdesc = "IRC bouncer"
+maintainer = "ttyyls <contact@behri.org>"
+license = "AGPL-3.0-or-later"
+url = "https://soju.im"
+source = f"https://git.sr.ht/~emersion/soju/archive/v{pkgver}.tar.gz"
+sha256 = "089b900ccd7fa009b7ae50b9c2b9e77099b50b2499422649c3b27aff7d62d109"
+env = {"GOFLAGS": "-buildmode=pie -tags=libsqlite3"}
+
+
+def post_prepare(self):
+    from cbuild.util import golang
+
+    golang.Golang(self).mod_download()
+
+
+def init_build(self):
+    from cbuild.util import golang
+
+    self.make_env.update(golang.get_go_env(self))
+
+
+def do_check(self):
+    from cbuild.util import golang
+
+    golang.Golang(self).check()
+
+
+def post_build(self):
+    self.make_build_args = [
+        "./contrib/migrate-db",
+        "./contrib/migrate-logs",
+        "./contrib/znc-import",
+    ]
+    from cbuild.util import golang
+
+    golang.Golang(self).build(args=["-buildmode=pie"])
```

Comment:
if you have to pass this then something else is wrong as that should be the default

probably you have to init self.golang = in init_build or wherever so everything is kept between the steps

Reply:
no, that seems to have been unecessary